### DISCUSSION TITLE
How to implement useHierarchicalMenu?

### DISCUSSION THREAD

**Discussion by recervictory** [algolia/instantsearch#5342] [Q&A] [answered]
I am struggling with react-instantsearch-hooks ? Can anyone tell me how to implement `useHierarchicalMenu`.
I am only getting level 0 as items of childeren data return null

**Comment by Haroenv** [comment, marked as answer]
you only receive the child items once there's a refinement on a parent level, as otherwise this would require a waterfall of network requests
